src/rmarkdown/knit.ts

```diff
@@ -0,0 +1,250 @@
+				`rmarkdown::render_site(${docPath})`;
+		} else {
+			knitCommand = outputFormat ?
+				`rmarkdown::render(${docPath}, output_format = '${outputFormat}')` :
+				`rmarkdown::render(${docPath})`;
+		}
+
+		return knitCommand.replace(/['"]/g, '\\"');
+	}
+
+	// check if the workspace of the document is a R Markdown site.
+	// the definition of what constitutes an R Markdown site differs
+	// depending on the type of R Markdown site (i.e., "simple" vs. blogdown sites)
+	private async findSiteParam(): Promise<string|undefined> {
+		const rootFolder = vscode.workspace.workspaceFolders[0].uri.fsPath;
+		const wad = vscode.window.activeTextEditor.document.uri.fsPath;
+		const indexFile = (await vscode.workspace.findFiles(new vscode.RelativePattern(rootFolder, 'index.{Rmd,rmd, md}'), null, 1))?.[0];
+		const siteRoot = path.join(path.dirname(wad), '_site.yml');
+
+		// 'Simple' R Markdown websites require all docs to be in the root folder
+		if (fs.existsSync(siteRoot)) {
+			return 'rmarkdown::render_site';
+		// Other generators may allow for docs in subdirs
+		} else if (indexFile) {
+			const indexData = this.getYamlFrontmatter(indexFile.fsPath);
+			if (indexData?.['site']) {
+				return indexData['site'];
+			}
+		}
+
+		return undefined;
+	}
+
+	// readme files should not be knitted via render_site
+	private isREADME(docPath: string) {
+		return !!path.basename(docPath).includes('README');
+	}
+
+	// alters the working directory for evaluating chunks
+	public setKnitDir(): void {
+		const currentDocumentWorkspacePath: string = vscode.workspace.getWorkspaceFolder(vscode.window.activeTextEditor?.document?.uri)?.uri?.fsPath;
+		const currentDocumentFolderPath: string = path.dirname(vscode.window?.activeTextEditor.document?.uri?.fsPath);
+		const items: IKnitQuickPickItem[] = [];
+
+		if (currentDocumentWorkspacePath) {
+			items.push(
+				{
+					label: (knitDir === KnitWorkingDirectory.workspaceRoot ? '$(check)' : '') + KnitWorkingDirectory.workspaceRoot,
+					value: KnitWorkingDirectory.workspaceRoot,
+					detail: 'Use the workspace root as the knit working directory',
+					description: currentDocumentWorkspacePath ?? currentDocumentFolderPath ?? 'No available workspace'
+				}
+			);
+		}
+
+		if (currentDocumentFolderPath && currentDocumentFolderPath !== '.') {
+			items.push(
+				{
+					label: (knitDir === KnitWorkingDirectory.documentDirectory ? '$(check)' : '') + KnitWorkingDirectory.documentDirectory,
+					value: KnitWorkingDirectory.documentDirectory,
+					detail: 'Use the document\'s directory as the knit working directory',
+					description: currentDocumentFolderPath ?? 'No folder available'
+
+				}
+			);
+		}
+
+		if (items.length > 0) {
+			void vscode.window.showQuickPick(
+				items,
+				{
+					title: 'Set knit working directory',
+					canPickMany: false,
+					onDidSelectItem: (item: IKnitQuickPickItem) => {
+						knitDir = item.value;
+					}
```

Comment:
`onDidSelectItem` triggers everytime you change the selection in the quickpick, e.g. using the arrow keys.
This leads to wrong behaviour, e.g. when the user highlights an option and then presses escape.

A better way is to use the return value of `showQuickPick` (which needs to be checked for `undefined` in case the user presses escape)